src/Disks/IO/ReadBufferFromRemoteFSGather.cpp

```diff
@@ -38,31 +38,31 @@ ReadBufferFromRemoteFSGather::ReadBufferFromRemoteFSGather(
         && (!FileCache::isReadOnly() || settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache);
 }
 
-SeekableReadBufferPtr ReadBufferFromRemoteFSGather::createImplementationBuffer(const String & path, size_t file_size)
+SeekableReadBufferPtr ReadBufferFromRemoteFSGather::createImplementationBuffer(const StoredObject & object)
 {
-    if (!current_file_path.empty() && !with_cache && enable_cache_log)
+    if (current_object && !with_cache && enable_cache_log)
     {
         appendFilesystemCacheLog();
     }
 
-    current_file_path = path;
-    current_file_size = file_size;
+    current_object = object;
     total_bytes_read_from_current_file = 0;
+    const auto & object_path = object.absolute_path;
 
-    size_t current_read_until_position = read_until_position ? read_until_position : file_size;
-    auto current_read_buffer_creator = [path, current_read_until_position, this]() { return read_buffer_creator(path, current_read_until_position); };
+    size_t current_read_until_position = read_until_position ? read_until_position : object.bytes_size;
+    auto current_read_buffer_creator = [=, this]() { return read_buffer_creator(object_path, current_read_until_position); };
```

Comment:
Maybe specify the capture list explicitly?

Reply:
I think current ways is more compact and nice + the lambda is just one line and captures two objects, so one will not have difficulties seeing what is captured + non explicit capture will capture only actually used variables, e.g. no difference as without explicit capture